File: src/main/java/cookiedragon/luchadora/Luchadora.java
package cookiedragon.luchadora;

import cookiedragon.luchadora.managers.*;
import cookiedragon.luchadora.module.impl.ui.HudManager;
import cookiedragon.luchadora.util.Initialisable;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.Mod;

@Mod(
	modid = Luchadora.MOD_ID,
	name = Luchadora.MOD_NAME,
	version = Luchadora.VERSION
)
public class Luchadora
{
	public static final String MOD_ID = "luchadora";
	public static final String MOD_NAME = "Luchadora";
	public static final String VERSION = "0.1";
	
	@Mod.Instance(MOD_ID)
	public static Luchadora INSTANCE;
	
	@Mod.EventHandler
	public void preInit(FMLPreInitializationEvent event)
	{
	}
	
	@Mod.EventHandler
	public void init(FMLInitializationEvent event)
	{
		for (Initialisable initialisable : new Initialisable[]{
			BindManager.INSTANCE,
			ModuleManager.INSTANCE,
			GenericEventListener.INSTANCE,
			ForgeEventListener.INSTANCE,
			PerspectiveManager.INSTANCE
		}) {
			initialisable.init();
		}
		HudManager.init();
		new DiscordIntegration().start();
	}
	
	@Mod.EventHandler
	public void postInit(FMLPostInitializationEvent event)
	{
	}
	
	public static String getBrand()
	{
		return String.format("%s %s", MOD_NAME, VERSION);
	}
}

File: src/main/kotlin/cookiedragon/luchadora/Luchadora.kt
package cookiedragon.luchadora

import cookiedragon.luchadora.integrations.discord.DiscordIntegration
import cookiedragon.luchadora.managers.*
import cookiedragon.luchadora.module.impl.ui.HudManager
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import java.awt.Frame
import javax.swing.JOptionPane

/**
 * @author cookiedragon234 17/Feb/2020
 */
@Mod(modid = Luchadora.MOD_ID, name = Luchadora.MOD_NAME, version = Luchadora.VERSION)
class Luchadora {
	@Mod.EventHandler
	fun preInit(event: FMLPreInitializationEvent) {
	}
	
	@Mod.EventHandler
	fun init(event: FMLInitializationEvent) {
		HudManager.init()
		for (toInit in arrayOf(
				BindManager,
				ModuleManager,
				GenericEventListener,
				ForgeEventListener,
				PerspectiveManager,
				DiscordIntegration
		)) { toInit.init() }
	}
	
	@Mod.EventHandler
	fun postInit(event: FMLPostInitializationEvent) {
	}
	
	companion object {
		const val MOD_ID = "luchadora"
		const val MOD_NAME = "Luchadora"
		const val VERSION = "0.1"
		@Mod.Instance(MOD_ID)
		var INSTANCE: Luchadora? = null
		
		@JvmStatic
		val brand: String
			get() = String.format("%s %s", MOD_NAME, VERSION)
	}
}
